chain/network/src/snapshot_hosts/mod.rs

```diff
@@ -331,46 +312,27 @@ impl SnapshotHostsCache {
         self.0.lock().hosts.iter().map(|(_, v)| v.clone()).collect()
     }
 
-    // Selecs a peer to send the request for this part ID to. Chooses based on a priority score
-    // calculated as a hash of the Peer ID plus the part ID, and will return different hosts
-    // on subsequent calls, eventually iterating over all valid SnapshotHostInfos we know about
-    // TODO: get rid of the dead_code and hook this up to the decentralized state sync
+    /// Given a state part request, selects a peer host to which the request should be sent.
     #[allow(dead_code)]
-    pub fn select_host(
+    pub fn select_host_for_part(
         &self,
         sync_hash: &CryptoHash,
         shard_id: ShardId,
-        part_id: &PartId,
+        part_id: u64,
     ) -> Option<PeerId> {
-        let mut inner = self.0.lock();
-        let num_hosts = inner.hosts.len();
-        let selector = inner.state_part_selectors.entry(shard_id).or_default();
-
-        if selector.tried_everybody(part_id) && selector.len(part_id) < num_hosts {
```

Comment:
This was a little wonky before since not all hosts can serve all parts, hence this condition can be true without there being any new entries to be inserted into the selector. Not a correctness issue but a small performance thing. By tracking the host count per shard we can handle this efficiently.